dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPIImplTest.java

```diff
@@ -387,13 +381,93 @@ public void publish(final TestCase testCase) throws DotPublishingException, DotS
+        extractTarArchive(bundleRoot, extractHere);
+        assertBundle(testAsset, dependencies, extractHere);
+    }
+
+    /**
+     * Method to Test: {@link PublisherAPIImpl#publish(PublisherConfig)}
+     * When: Add different assets into a bundle, and send it
+     * Should: Create all the files
+     */
+    @Test
+    @UseDataProvider("publishers")
+    public void sendPushPublishBundle(final TestAsset testAsset)
+            throws DotPublishingException, DotSecurityException, IOException, DotDataException, DotPublisherException {
+        final Class<? extends Publisher> publisher = PushPublisher.class;
+
+        final Environment environment = new EnvironmentDataGen().nextPersisted();
+
+        final PushPublishingEndPoint publishingEndPoint = new PushPublishingEndPointDataGen()
+                .environment(environment)
+                .nextPersisted();
+
+        final FilterDescriptor filterDescriptor = new FilterDescriptorDataGen().nextPersisted();
+
+        final PushPublisherConfig config = new PushPublisherConfig();
+        config.setPublishers(list(publisher));
+        config.setOperation(PublisherConfig.Operation.PUBLISH);
+        config.setLuceneQueries(list());
+        config.setId("sendPushPublishBundle_" + System.currentTimeMillis());
+
+        final Bundle bundle = new BundleDataGen()
+                .pushPublisherConfig(config)
+                .addAssets(list(testAsset.asset))
+                .filter(filterDescriptor)
+                .nextPersisted();
+
+        final BundleFactoryImpl bundleFactory = new BundleFactoryImpl();
+        bundleFactory.saveBundleEnvironment(bundle, environment);
+
+        final Collection<Object> dependencies = new HashSet<>();
+        dependencies.addAll(testAsset.expectedInBundle);
+
+        createLanguageVariableIfNeeded();
+        addLanguageVariableDependencies(dependencies, testAsset.addLanguageVariableDependencies);
+
+        final PublisherAPIImpl publisherAPI = new PublisherAPIImpl();
+
+        final PublishAuditStatus publishAuditStatus = new PublishAuditStatus(bundle.getId());
+
+        final PublishAuditHistory publishAuditHistory = new PublishAuditHistory();
+        publishAuditStatus.setStatusPojo(publishAuditHistory);
+
+        PublishAuditAPI.getInstance().insertPublishAuditStatus(publishAuditStatus);
+
+        final File tempFile = com.dotmarketing.util.FileUtil
+                .createTemporaryFile("sendPushPublishBundle_");
+
+        HttpServer httpServer = createHttpServer(tempFile);
+
+        try {
+            httpServer.start();
+            final PublishStatus publish = publisherAPI.publish(config);
+            File bundleRoot = publish.getOutputFiles().get(0);
+
+            assertTrue(tempFile.exists());
+            assertTrue(tempFile.length() > 0);
+
             final File extractHere = new File(bundleRoot.getParent() + File.separator + config.getName());
             extractTarArchive(bundleRoot, extractHere);
```

Comment:
should be tempFile instead of bundleRoot so we checked in received tar

Reply:
done https://github.com/dotCMS/core/pull/20428/commits/a7c1a059c12e8a2d8561110fc166fc5b7db6af8c#diff-81d6bc992715805cfe2e2bee2c1eeaa4ca1000153e04e79b682d47084d3da682R452